### PR TITLE
Feature/ranking list

### PR DESCRIPTION
**Pull request by Julian0701**
### DEVELOP
- [x] leaderboard.tsx
- [x] board_page_body.tsx
- [x] user_personal_ranking.tsx
- [x] leaderboard_tab.tsx

### CHECK LIST
- [Naming convention](https://github.com/CAFECA-IO/WorkGuidelines/blob/main/technology/coding-convention/naming-convention.md) verification: checked
- Coding style verification: checked
- new Library: 0
- new Class / Component: 5
    - src/components/leaderboard_tab/leaderboard_tab.tsx
    - src/constants/ranking_time_span.tsx
    - src/interfaces/tidebit_defi_background/user_assets.ts
    - src/interfaces/tidebit_defi_background/personal_ranking.ts
    - src/interfaces/tidebit_defi_background/leaderboard.ts
- new loop: 6
  - src/lib/common.ts line295
  - src/lib/common.ts line296
  - src/components/leaderboard_tab/leaderboard_tab.tsx line148
  - src/components/leaderboard_tab/leaderboard_tab.tsx line160
  - src/components/leaderboard_tab/leaderboard_tab.tsx line219
  - src/components/leaderboard_tab/leaderboard_tab.tsx line234
- new recursive: 0
- high risk: 0
- new sql: 0

### UML

- [Sequence Diagram 001 (interaction between user, frontend and backend)](https://github.com/CAFECA-IO/Documents/blob/main/TBD/TBDSD00001.md)
- [Sequence Diagram 002 (background mechanism)](https://github.com/CAFECA-IO/Documents/blob/main/TBD/TBDSD00002.md)
- [Component Diagram](https://github.com/CAFECA-IO/Documents/blob/main/TBD/TBDCP00001.md)
- [Class Diagram](https://github.com/CAFECA-IO/Documents/blob/main/TBD/TBDCD00001.md)
- [Global context document](https://www.notion.so/Global-Context-public-d93ac7c941c94fb6b173d34a7bc4e959)
- [Table of interface](https://github.com/CAFECA-IO/EventReport/blob/main/meeting/20230106_tbd_mvp.md)

### CODE REVIEW

File: src/contexts/user_context.tsx [hunk 1]
```diff
@@ -813,6 +822,16 @@ export const UserProvider = ({children}: IUserProvider) => {
     return myAsset;
   };
 
+  /* ToDo: (20230510 - Julian) get data from backend */
+  const getMyRanking = (currency: string) => {
```
Comment (Julian0701) [inline src/contexts/user_context.tsx:826]: live/daily/weekly/monthly

File: src/contexts/market_context.tsx [hunk 2]
```diff
@@ -419,6 +429,120 @@ export const MarketProvider = ({children}: IMarketProvider) => {
     return result;
   };
 
+  // TODO: (20230511 - Julian) get data from backend
+  const getLeaderboard = (timeSpan: IRankingTimeSpan) => {
+    let result: ILeaderboard[] = [];
```
Comment (Julian0701) [inline src/contexts/market_context.tsx:434]: interface

File: src/components/leaderboard_tab/leaderboard_tab.tsx [hunk 3]
```diff
@@ -0,0 +1,249 @@
...
+
+const LeaderboardTab = () => {
+  const marketCtx = useContext(MarketContext);
+  const windowSize = useWindowSize();
+
+  const podiumWidth =
+    windowSize.width > DEFAULT_PODIUM_WIDTH ? windowSize.width : DEFAULT_PODIUM_WIDTH;
+  const podiumHeight = (DEFAULT_PODIUM_HEIGHT / DEFAULT_PODIUM_WIDTH) * podiumWidth;
+  const medalistSize =
+    (podiumWidth - 500) / 3 < DEFAULT_MEDALIST_SIZE
```
Comment (Julian0701) [inline src/components/leaderboard_tab/leaderboard_tab.tsx:23]: contants

File: src/contexts/user_context.tsx [hunk 4]
```diff
@@ -813,6 +823,47 @@ export const UserProvider = ({children}: IUserProvider) => {
     return myAsset;
   };
 
+  /* ToDo: (20230510 - Julian) get data from backend */
+  const getMyRanking = (timeSpan: IRankingTimeSpan) => {
+    let myRanking: IMyRanking = {
```
Comment (Julian0701) [inline src/contexts/user_context.tsx:828]: interface

**Review by Luphia — APPROVED** [review]
- no bug found

**Pull request merged**